src/confirmations_impl.h

```diff
@@ -2,98 +2,199 @@
+  void AdSustained(std::unique_ptr<NotificationInfo> info) override;
   void OnTimer(const uint32_t timer_id) override;
 
  private:
   bool is_initialized_;
+  bool is_wallet_initialized_;
+  bool is_issuers_initialized_;
+
+  const size_t low_token_threshold = 20;
+  const size_t refill_amount = 5 * low_token_threshold;
+
+  WalletInfo wallet_info_;
+
+  uint32_t step_2_refill_confirmations_timer_id_;
+  void StartRefillingConfirmations(const uint64_t start_timer_in);
+  void RefillConfirmations();
+  void StopRefillingConfirmations();
+  bool IsRefillingConfirmations() const;
+
+  uint32_t step_4_retrieve_payment_ious_timer_id_;
+  void StartRetrievingPaymentIOUsTimer();
+  void StartRetrievingPaymentIOUs(const uint64_t start_timer_in);
+  void RetrievePaymentIOUs();
+  void StopRetrievingPaymentIOUs();
+  bool IsRetrievingPaymentIOUs() const;
+
+  uint32_t step_5_cash_in_payment_ious_timer_id_;
+  void StartCashingInPaymentIOUs(const uint64_t start_timer_in);
+  void CashInPaymentIOUs();
+  void StopCashingInPaymentIOUs();
+  bool IsCashingInPaymentIOUs() const;
+
+  uint32_t fetch_tokens_timer_id_;
+  std::string fetch_tokens_server_url_;
+  void StartFetchingTokens(const uint64_t start_timer_in);
+  void FetchTokens();
+  void StopFetchingTokens();
+  bool IsFetchingTokens() const;
+
+  /////////////////////////////////////////////////////////////////////////////
+  // persist these properties
+
+  // If unset or "0", assume we haven't gotten one
+  std::string issuers_version_ = "0";
+
+  // If this changes what we can do is burn .*confirmation_tokens.* & repop
+  std::string server_confirmation_key_;
+
+  std::vector<std::string> server_bat_payment_names;
+  std::vector<std::string> server_bat_payment_keys;
+
+  std::vector<std::string> original_confirmation_tokens;
+  std::vector<std::string> blinded_confirmation_tokens;
+  std::vector<std::string> signed_blinded_confirmation_tokens;
+
+  std::vector<std::string> payment_token_json_bundles;
+  std::vector<std::string> signed_blinded_payment_token_json_bundles;
+  std::vector<std::string> fully_submitted_payment_bundles;
+
+  /////////////////////////////////////////////////////////////////////////////
+
+  std::string real_wallet_address_;
+  std::string real_wallet_address_secret_key_;
+  std::string local_server_confirmation_key_;
+  std::vector<std::string> local_original_confirmation_tokens_;
+  std::vector<std::string> local_blinded_confirmation_tokens_;
+  std::string confirmation_id_;
+  std::string local_original_payment_token_;
+  std::string local_blinded_payment_token_;
+  std::string blinded_payment_token_;
+  base::DictionaryValue* map_;
+  std::string bundle_json_;
+
+  /////////////////////////////////////////////////////////////////////////////
+
+  void SetConfirmationsStatus();
+
+  std::string GetServerUrl();
+  int GetServerPort();
+
+  void Step1StoreTheServersConfirmationsPublicKeyAndGenerator(
+      std::string confirmations_GH_pair,
```

Comment:
Should this be const?

Reply:
This library is not const correct, but I did not want to change original code, so ok to fix later?